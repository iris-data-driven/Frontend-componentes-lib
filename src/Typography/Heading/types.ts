type HeadingTypes = 'h1' | 'h2' | 'h3';

export type StyledHeadingProps = HeadingProps & {
  as: HeadingTypes;
};

export type HeadingProps = {
  type: HeadingTypes;
  className?: string;
};
