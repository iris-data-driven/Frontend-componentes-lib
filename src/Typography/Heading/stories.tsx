import React from 'react';

import Heading from './';

export default {
  title: 'Typography|Heading',
};

export const Heading1 = () => <Heading type="h1">Heading 1</Heading>;
export const Heading2 = () => <Heading type="h2">Heading 2</Heading>;
export const Heading3 = () => <Heading type="h3">Heading 3</Heading>;
