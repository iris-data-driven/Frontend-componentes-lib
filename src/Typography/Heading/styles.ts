import styled from 'styled-components';

import texts from 'styles/texts';

import { StyledHeadingProps } from './types';

export const Wrapper = styled.h1<StyledHeadingProps>`
  ${props => texts.headings[props.type]}
`;
