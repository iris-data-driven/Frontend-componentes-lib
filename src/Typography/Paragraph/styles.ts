import styled from 'styled-components';

import texts from 'styles/texts';

import { ParagraphProps } from './types';

export const Wrapper = styled.p<ParagraphProps>`
  ${props => texts.paragraphs[props.type]}
`;
