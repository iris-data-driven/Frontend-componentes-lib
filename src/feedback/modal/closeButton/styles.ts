import styled from 'styled-components';

import Close from 'icons/Close';

export const Wrapper = styled.div`
  display: flex;
  justify-content: flex-end;
  padding: 8px 16px;
`;

export const CloseIcon = styled(Close).attrs(() => ({
  role: 'button',
  'aria-label': 'close modal',
}))`
  :hover {
    cursor: pointer;
  }
  width: 19px;
  height: 19px;
`;
