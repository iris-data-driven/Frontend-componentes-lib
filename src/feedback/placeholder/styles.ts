import styled, { css } from 'styled-components';

import texts from 'styles/texts';

import { PlaceholderStyleProps } from './types';

const textSytle = css`
  width: 250px;
  height: ${texts.sizes.paragraph.text1};
`;

const headingSytle = css`
  width: 170px;
  height: ${texts.sizes.heading.h1};
`;

const circleStyle = css`
  border-radius: 50%;
  width: 70px;
  height: 70px;
`;

const placeholderStyles = {
  text: textSytle,
  heading: headingSytle,
  circle: circleStyle,
};

export const Wrapper = styled.div<PlaceholderStyleProps>`
  display: block;
  background-color: #b2b2b2;
  border-radius: 20px;

  ${props => placeholderStyles[props.type]}

  animation: react-placeholder-pulse 1.5s infinite;

  @keyframes react-placeholder-pulse {
    0% {
      opacity: 0.6;
    }
    50% {
      opacity: 1;
    }
    100% {
      opacity: 0.6;
    }
  }
`;

export const CustomPlaceholder = styled(Wrapper)`
  width: 100%;
  height: 100px;
`;
