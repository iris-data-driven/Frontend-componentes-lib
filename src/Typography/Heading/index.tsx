import React, { FC } from 'react';

import { Wrapper } from './styles';

import { HeadingProps } from './types';

const Heading: FC<HeadingProps> = ({ type, children }) => (
  <Wrapper data-testid="heading" type={type} as={type}>
    {children}
  </Wrapper>
);

export default Heading;
