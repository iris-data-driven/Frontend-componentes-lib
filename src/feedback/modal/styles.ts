import styled from 'styled-components';

import ReactModal from 'react-modal';

import { colors } from 'styles';

const backgroundWithOpacity = colors.neutral.black + '1a';

const sizes = {
  medium: {
    width: '590px',
    height: '335px',
  },
  large: {
    width: '655px',
    height: '370px',
  },
};

const sizes = {
  medium: {
    width: '590px',
    height: '335px',
  },
  large: {
    width: '655px',
    height: '370px',
  },
};

export const modalStyle = size => ({
  content: {
    backgroundColor: '#fff',
    opacity: 1,
    ...sizes[size],
  },
  overlay: {
    position: 'fixed',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: '100vh',
    width: '100vw',
    backgroundColor: backgroundWithOpacity,
  },
});

export const Wrapper = styled.div``;

export const Modal = styled(ReactModal)`
  height: 300px;
  width: 450px;
`;
