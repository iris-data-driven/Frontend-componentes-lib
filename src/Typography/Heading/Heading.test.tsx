import React from 'react';
import { render, screen } from '@testing-library/react';
import 'jest-styled-components';

import Heading from './';

import { texts } from '../../styles';

describe('Heading', () => {
  it('Should render the heading text correctly', () => {
    const text = 'This is a heading type 1';

    render(<Heading type="h1">{text}</Heading>);

    const component = screen.getByTestId('heading');
    expect(component).toHaveTextContent(text);
    expect(component).toMatchSnapshot();
    expect(component).toHaveStyleRule('color', '#333');
    expect(component).toHaveStyleRule('font-family', "'Raleway Bold'");
  });
});
