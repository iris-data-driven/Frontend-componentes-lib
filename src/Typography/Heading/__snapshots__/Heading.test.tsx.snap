// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Should render the heading text correctly 1`] = `
.c0 {
  font-family: Raleway Bold;
  font-style: normal;
  font-size: 25px;
  line-height: calc(25px * 1.5);
  color: #333;
}

<h1
  class="c0"
  data-testid="heading"
  type="h1"
>
  This is a heading type 1
</h1>
`;
